src/components/Asset/AssetActions/Compute/index.tsx

```diff
@@ -1,232 +1,286 @@
-  FileInfo
+  FileInfo,
+  Datatoken,
+  ProviderInstance,
+  ComputeAsset,
+  ZERO_ADDRESS,
+  ComputeEnvironment,
+  LoggerInstance,
+  ComputeAlgorithm,
+  ComputeOutput,
+  ProviderComputeInitializeResults,
+  unitsToAmount
 } from '@oceanprotocol/lib'
 import { toast } from 'react-toastify'
 import Price from '@shared/Price'
 import FileIcon from '@shared/FileIcon'
 import Alert from '@shared/atoms/Alert'
 import { useWeb3 } from '@context/Web3'
-import {
-  generateBaseQuery,
-  getFilterTerm,
-  queryMetadata
-} from '@utils/aquarius'
 import { Formik } from 'formik'
 import { getInitialValues, validationSchema } from './_constants'
-import axios from 'axios'
 import FormStartComputeDataset from './FormComputeDataset'
 import styles from './index.module.css'
 import SuccessConfetti from '@shared/SuccessConfetti'
-import { getServiceByName } from '@utils/ddo'
+import { getServiceByName, secondsToString } from '@utils/ddo'
+import {
+  isOrderable,
+  getAlgorithmAssetSelectionList,
+  getAlgorithmsForAsset,
+  getComputeEnviroment
+} from '@utils/compute'
 import { AssetSelectionAsset } from '@shared/FormFields/AssetSelection'
 import AlgorithmDatasetsListForCompute from './AlgorithmDatasetsListForCompute'
-import { getPreviousOrders } from '@utils/subgraph'
 import AssetActionHistoryTable from '../AssetActionHistoryTable'
 import ComputeJobs from '../../../Profile/History/ComputeJobs'
 import { useCancelToken } from '@hooks/useCancelToken'
-import { useIsMounted } from '@hooks/useIsMounted'
-import { SortTermOptions } from '../../../../@types/aquarius/SearchQuery'
-import { getAccessDetails } from '@utils/accessDetailsAndPricing'
-import { AccessDetails } from 'src/@types/Price'
-import { transformAssetToAssetSelection } from '@utils/assetConvertor'
+import { Decimal } from 'decimal.js'
+import { useAbortController } from '@hooks/useAbortController'
+import { getOrderPriceAndFees } from '@utils/accessDetailsAndPricing'
+import { OrderPriceAndFees } from 'src/@types/Price'
+import { handleComputeOrder } from '@utils/order'
+import { AssetExtended } from 'src/@types/AssetExtended'
+import { getComputeFeedback } from '@utils/feedback'
+import { usePool } from '@context/Pool'
 import { useMarketMetadata } from '@context/MarketMetadata'
+import { getPoolData } from '@context/Pool/_utils'
+import { getDummyWeb3 } from '@utils/web3'
+import { initializeProviderForCompute } from '@utils/provider'
 
 export default function Compute({
-  ddo,
-  accessDetails,
+  asset,
   dtBalance,
   file,
   fileIsLoading,
-  isConsumable,
   consumableFeedback
 }: {
-  ddo: Asset
-  accessDetails: AccessDetails
+  asset: AssetExtended
   dtBalance: string
   file: FileInfo
   fileIsLoading?: boolean
-  isConsumable?: boolean
   consumableFeedback?: string
 }): ReactElement {
-  const { appConfig } = useMarketMetadata()
-  const { accountId } = useWeb3()
+  const { accountId, web3 } = useWeb3()
+  const { getOpcFeeForToken } = useMarketMetadata()
```

Comment:
why do we have 2 hooks doing exactly the same thing, but named differently, and both used in this file?

Reply:
wrong line, I mean the `useAbortController()` and `useCancelToken()` hooks which are both in use in here. axios `CancelToken` is deprecated so only `useAbortController()` should be used in new code

Reply:
sorry, didn't see the chain of events ends up with `queryMetadata()` requiring `CancelToken` so changing this there to `AbortSignal` has far reaching consequences in whole codebase which we should not do in here